cirq-ionq/cirq_ionq/ionq_native_gates_test.py

```diff
@@ -74,3 +86,50 @@ def test_ms_unitary(phases):
 
     mat = cirq.protocols.unitary(gate)
     numpy.testing.assert_array_almost_equal(mat.dot(mat.conj().T), numpy.identity(4))
+
+
+@pytest.mark.parametrize(
+    "gate",
+    [
+        *[ionq.GPIGate(phi=angle) for angle in PARAMS_FOR_ONE_ANGLE_GATE],
+        *[ionq.GPI2Gate(phi=angle) for angle in PARAMS_FOR_ONE_ANGLE_GATE],
+        *[ionq.MSGate(phi0=angles[0], phi1=angles[1]) for angles in PARAMS_FOR_TWO_ANGLE_GATE],
+    ],
+)
+def test_gate_inverse(gate):
+    """Tests that the inverse of natives gate are correct."""
+    mat = cirq.protocols.unitary(gate)
+    mat_inverse = cirq.protocols.unitary(gate**-1)
+    num_qubits = mat.shape[0]
```

Comment:
Actually, this isn't the number of qubits. It's the Hilbert space dimension, which is two to the power number of qubits. Perhaps

```python
dim = mat.shape[0]
```

Reply:
thanks for the catch! Fixed.